Library/Formula/osquery.rb

```diff
@@ -66,7 +66,8 @@ def install
       end
     end
 
-    system "cmake", ".", *std_cmake_args
+    # Append libboost_thread-mt since osquery expects cpp-netlib-0.12.
+    system "cmake", ".", *std_cmake_args, "-DCMAKE_EXE_LINKER_FLAGS=-lboost_thread-mt"
```

Comment:
See the discussion from #50362.

To support cpp-netlib-0.11 we need to explicitly add a dependent link to `boost_thread`. When cpp-netlib-0.12 is released we can modify the dependencies to include `asio`, bump the package revision, and remove this explicit linker addition.